crypto/ex_data.c

```diff
@@ -287,8 +287,6 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         CRYPTOerr(CRYPTO_F_CRYPTO_DUP_EX_DATA, ERR_R_MALLOC_FAILURE);
         return 0;
     }
-    if (!CRYPTO_set_ex_data(to, mx - 1, NULL))
-        goto err;
```

Comment:
PR #2988 by @bernd-edlinger added these lines to set the last ex_data item to NULL (which can cause a memory leak, among other bad things). This PR undoes that change, but I'd like to know what the intention of adding this was to make sure the correct thing is done here.